controllers/finalizers.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	replicationv1alpha1 "github.com/csi-addons/volume-replication-operator/api/v1alpha1"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	volumeReplicationFinalizer = "replication.storage.openshift.io"
+)
+
+// addFinalizerToVR adds the VR finalizer on the VolumeReplication instance.
+func (r *VolumeReplicationReconciler) addFinalizerToVR(logger logr.Logger, vr *replicationv1alpha1.VolumeReplication,
```

Comment:
Suggest to leverate controllerutil [abstraction](https://github.com/kubernetes-sigs/controller-runtime/blob/1e4d87c9f9e15e4a58bb81909dd787f30ede7693/pkg/finalizer/finalizer.go#L50) for the same purpose, if applicable.

Reply:
will try to see if fits, if this fits will consider as an enhancement PR. Thanks for the suggestion @ShyamsundarR